requre/helpers/httpx_response.py

```diff
@@ -0,0 +1,265 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+
+import datetime
+import json
+from contextlib import contextmanager
+from typing import Any, Dict, List, Optional
+
+import httpx
+
+from requre.cassette import Cassette
+from requre.objects import ObjectStorage
+from requre.record_and_replace import make_generic, recording, replace
+
+
+class HTTPXRequestResponseHandling(ObjectStorage):
+    __response_keys = ["status_code", "encoding"]
+    __ignored = ["cookies"]
+    __response_keys_special = ["next_request", "headers", "_elapsed", "_content"]
+    __store_indicator = "__store_indicator"
+    __implicit_encoding = "utf-8"
+
+    def __init__(
+        self,
+        store_keys: list,
+        cassette: Optional[Cassette] = None,
+        response_headers_to_drop=None,
+    ) -> None:
+        # replace request if given as key and use prettier url
+        for index, key in enumerate(store_keys):
+            if isinstance(key, httpx.Request):
+                store_keys[index] = str(key.url)
+                store_keys.insert(index, key.method)
+        super().__init__(store_keys, cassette=cassette)
+        self.response_headers_to_drop = response_headers_to_drop or []
+
+    def write(
+        self, response: httpx.Response, metadata: Optional[Dict] = None
+    ) -> httpx.Response:
+        super().write(response, metadata)
+        # TODO: disabled for now, improve next handling if we find it makes sense
+        # if getattr(response, "next"):
+        #    self.write(getattr(response, "next"))
+        return response
+
+    def read(self):
+        data = super().read()
+        # TODO: disabled for now, improve next handling if we find it makes sense
+        # if getattr(data, "next"):
+        #    data._next = self.read()
+        return data
+
+    def to_serializable(self, response: httpx.Response) -> Any:
+        output = dict()
+        for key in self.__response_keys:
+            output[key] = getattr(response, key)
+        for key in self.__response_keys_special:
+            if key == "headers":
+                headers_dict = dict(response.headers)
+                for header in self.response_headers_to_drop:
+                    if header in headers_dict:
+                        headers_dict[header] = None
+                output[key] = headers_dict
+            if key == "_elapsed":
+                output[key] = response.elapsed.total_seconds()
```

Comment:
Do we need this? Can't this make it hard to re-generate with small/no diff?

Reply:
There is a requre hook for resetting this to a fixed value…